### PR TITLE
PPX memory interface

### PR DESCRIPTION
**Pull request by jonludlam**
Requires PRs in squeezed and xenopsd. This is the first interface to be changed, others are to come.

### CODE REVIEW

File: memory/memory_interface.ml [hunk 1]
```diff
@@ -14,47 +14,212 @@
...
+open Idl
 
 let service_name = "memory"
 let queue_name = Xcp_service.common_prefix ^ service_name
 let json_path = "/var/xapi/memory.json"
 let xml_path = "/var/xapi/memory"
 
-type reservation_id = string
+type reservation_id =
+  string [@@doc [
```
Comment (mseri) [inline memory/memory_interface.ml]: With documentation generation? :tada:

File: memory/memory_interface.ml [hunk 2]
```diff
@@ -14,47 +14,212 @@
...
+
+type reserve_memory_range_result = reservation_id * int64
+[@@deriving rpcty]
+
+module API(R : RPC) = struct
+  open R
+
+  let description = Interface.{
+      name = "Memory";
+      namespace = None;
```
Comment (mseri) [inline memory/memory_interface.ml]: What would be the effect of `Some "something"` here?

**Comment by robhoes** [comment]
The revolution is about to begin!

**Comment by mseri** [comment]
For reference: 

- squeezed PR https://github.com/xapi-project/squeezed/pull/53
- xenopsd PR https://github.com/xapi-project/xenopsd/pull/354

**Comment by jonludlam** [comment]
@mseri: You can certainly return different error types from different API calls - is that what you're after?

File: memory/memory_interface.ml (hunk 2 above)
Comment (jonludlam) [inline memory/memory_interface.ml]: It just changes the wire-name of the API call

**Comment by mseri** [comment]
Kind of. In that case I wanted to make sure that only apis in the `Nvidia` module could return the nvidia specific errors, but I also wanted to define them inside the `Nvidia` module and not at toplevel and this could not be done (to avoid accumulating NvmlError, AMDError, IntelError, ...)

**Comment by jonludlam** [comment]
Yes, you can declare those error types within the submodules.

**Pull request merged**